base/src/main/java/com/smartdevicelink/managers/screen/choiceset/BaseChoiceSetManager.java

```diff
@@ -541,6 +546,12 @@ public void onCapabilityRetrieved(Object capability) {
                         int currentWindowID = windowCapability.getWindowID() != null ? windowCapability.getWindowID() : PredefinedWindows.DEFAULT_WINDOW.getValue();
                         if (currentWindowID == PredefinedWindows.DEFAULT_WINDOW.getValue()) {
                             defaultMainWindowCapability = windowCapability;
+
+                            KeyboardCapabilities keyboardCapabilities = windowCapability.getKeyboardCapabilities();
+                            if (keyboardCapabilities.getMaskInputCharactersSupported() != null
+                                    && !keyboardCapabilities.getMaskInputCharactersSupported()) {
+                                keyboardConfiguration.setMaskInputCharacters(KeyboardInputMask.DISABLE_INPUT_KEY_MASK);
+                            }
```

Comment:
@kostyaBoss Do you need to update the unit tests?

Reply:
I don't think the `ChoiceSetManagerTests` is testing what's going on it it's `DisplayCapabilityListener` at least I didn't find it in the current tests and it seems to be never testing before. If we still need to add those tests - let me know